src/WebJobs.Script.WebHost/App_Start/WebHostSettings.cs

```diff
@@ -1,13 +1,28 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System.Runtime.Serialization;
+using Microsoft.Azure.WebJobs.Host;
+using Newtonsoft.Json;
+
 namespace Microsoft.Azure.WebJobs.Script.WebHost
 {
     public class WebHostSettings
     {
+        [JsonProperty("isSelfHost")]
```

Comment:
I think you can just use the existing HostStatus.Id property for your purposes and revert all these changes in this file.